test/test_autograd.py

```diff
@@ -2512,7 +2513,10 @@ def run_functional_checks(test_case, test_name, name, apply_fn, run_grad_checks,
 
     for dim_perm in product([-1, 1], repeat=len(dim_args_idx)):
         test_name = basic_test_name
-        new_args = [arg * dim_perm[dim_args_idx.index(i)] if i in dim_args_idx else arg for i, arg in enumerate(args)]
+        if args == NO_ARGS:
+            new_args = []
+        else:
+            new_args = [arg * dim_perm[dim_args_idx.index(i)] if i in dim_args_idx else arg for i, arg in enumerate(args)]
```

Comment:
Is this case necessary? It should be an empty list comprehension if `args` is an empty tuple

Reply:
when args == NO_ARGS enumerate doesn't work on it; I can add an iterator that immediately exits.

Reply:
What's the problem? I just tried enumerating an empty sequence and it was ok

Reply:
it's not an empty sequence, it's an object().